src/apps/intel_mp/intel_mp.lua

```diff
@@ -953,9 +1150,91 @@ function Intel82599:init ()
    self.r.CTRL_EXT:set(bits {NS_DIS = 1})
 
    self:rss_enable()
+
+   if self.vmdq then
+      self:vmdq_enable()
+   end
+
    self:unlock_sw_sem()
 end
 
+-- enable VMDq mode, see 4.6.10.1
+-- follows the configuration flow in 4.6.11.3.3
+-- (should only be called on the master instance)
+function Intel82599:vmdq_enable ()
+   -- must be set prior to setting MTQC (7.2.1.2.1)
+   self.r.RTTDCS:set(bits { ARBDIS=6 })
+
+   -- 1010 -> 32 pools, 4 RSS queues each
+   self.r.MRQC:bits(0, 4, 0xA)
+
+   -- TODO: not sure this is needed, but it's in intel10g
+   -- disable RSC (7.11)
+   self.r.RFCTL:set(bits { RSC_Dis=5 })
+
+   -- 128 Tx Queues, 64 VMs (4.6.11.3.3)
+   self.r.MTQC(bits { VT_Ena=1, Num_TC_OR_Q=2 })
+
+   -- enable virtualization, replication enabled, disable default pool
+   self.r.PFVTCTL(bits { VT_Ena=0, Rpl_En=30, DisDefPool=29 })
+
+   -- enable VMDq Tx to Rx loopback
+   self.r.PFDTXGSWC:set(bits { LBE=0 })
+
+   -- needs to be set for loopback (7.10.3.4)
+   self.r.FCRTH[0](0x10000)
+
+   -- enable vlan filter (4.6.7, 7.1.1.2)
+   self.r.VLNCTRL:set(bits { VFE=30 })
+
+   -- intel10g zeroes out ETQF,ETQS here but they are init to 0
+
+   -- RTRUP2TC/RTTUP2TC cleared above in init
+
+   -- DMA TX TCP max allowed size requests (set to 1MB)
+   self.r.DTXMXSZRQ(0xFFF)
+
+   -- disable PFC, enable legacy control flow
+   self.r.MFLCN(bits { RFCE=3 })
+   self.r.FCCFG(bits { TFCE=3 })
+
+   -- RTTDT2C, RTTPT2C, RTRPT4C cleared above in init()
+
+   -- QDE bit = 0 for all queues
+   for i = 0, 127 do
+      self.r.PFQDE(bor(lshift(1,16), lshift(i,8)))
+   end
```

Comment:
It's mentioned in 4.6.11.3.1:

> Queue Drop Enable (PFQDE) — In SR-IO the QDE bit should be set to 1b in the
PFQDE register for all queues. In VMDq mode, the QDE bit should be set to 0b for
all queues.

I think 4.6.11.3.1 is relevant since 4.6.11.3.3 says:

> Set the configuration bits as specified in Section 4.6.11.3.1 with the following
exceptions: